citelang/main/colors.py

```diff
@@ -24,6 +24,14 @@
 }
 
 
+# Generally safe colors for light or black backgrounds in the 16 color palette.
```

Comment:
Oh neat, this makes sense! What if the terminal has a light background? Maybe we could have sets for dark / light and then default to dark, allow a boolean or envar to indicate light (or I wonder if there is a more programmatic way to do it?)

Reply:
If you want me to do a PR to your PR, I can give this a shot. I think I also might print out the entire range of colors and choose a slightly larger set so it's more fun :)

Reply:
Hmm - I replied by email yesterday, but it didn't appear here... apologies.

The colors I've picked are in the basic 16 color terminal palette. These are always user-modifiable, and should be setup in terminal themes so that they are visible on whatever dark or light background the theme uses... Slight caveat that you need to avoid color 0/8 (generally black or close to it), and color 7/15 (generally white or close to it).

Historically the colors are (from Xterm)...

```
0 Black        8 Grey
1 Maroon       9 Red
2 Green       10 Lime
3 Olive       11 Yellow
4 Navy        12 Blue
5 Purple      13 Fuchsia
6 Teal        14 Aqua
7 Silver      15 White
```

... but they are generally customized a bit in terminal themes to be things that are a bit less garish.

If these colors (1-6, 9-14) are _not_ properly visible on your terminal, then the argument would be that your terminal's color theme is broken... so if you are using just these basic colors, then you don't really have to accommodate light/dark backgrounds specifically.

The need to accommodate light/dark backgrounds in your own code comes into play if you go outside of this range, and into the 256 color palette. That has fixed colors (outside of the first 16)... that aren't adapted by a user's terminal theme for visibility. So, you need to have a flag, like you say, to accommodate different themes that work on light / dark backgrounds. Or... some way of reading the background color that's in effect. applying minimum contrast etc.

Reply:
ah gotcha! So I tested locally, it looks okay :) Happy to merge.